# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/aniqaatiq842-commits/Flyrank-ML-INTERNSHIP/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My Lane

**Chosen Lane:** Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because it focuses on helping reviewers decide which content pages should be reviewed first. Instead of trying to automate content decisions, the goal is to build an evidence-based ranking that helps humans prioritize limited review time. This problem matches the starter dataset and allows me to compare a simple rule-based baseline with machine learning methods.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


# Research Question

### Search Question

Which content pages should be reviewed first for content refresh based on observable search and engagement signals?

### Decision

Decide which pages should be prioritized for review.

### Action

Create a ranked review queue so reviewers can focus on the highest-priority pages first.

### Cost of a Wrong Recommendation

If low-priority pages are reviewed first, valuable opportunities may be missed and reviewer time may be wasted. This does not necessarily mean a page would recover if refreshed, only that it deserves earlier human review.

### Why Data and Machine Learning Help

Many signals such as impressions, clicks, CTR, engagement rate, content age, freshness, and search position interact with each other. Machine learning can learn these relationships more effectively than fixed rule-based scoring while still supporting human decision-making.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
!git clone https://github.com/aniqaatiq842-commits/Flyrank-ML-INTERNSHIP.git

Cloning into 'Flyrank-ML-INTERNSHIP'...
remote: Enumerating objects: 111, done.
remote: Counting objects: 100% (111/111), done.
remote: Compressing objects: 100% (68/68), done.
remote: Total 111 (delta 28), reused 97 (delta 27), pack-reused 0 (from 0)
Receiving objects: 100% (111/111), 1.82 MiB | 10.17 MiB/s, done.
Resolving deltas: 100% (28/28), done.


In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("./Flyrank-ML-INTERNSHIP/data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))

df.head()

Rows: 30000
Columns: 44


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [9]:
print(df["trend_direction"].value_counts())

print("\nContent Types:")
print(df["content_type"].value_counts())

print("\nContent Age:")
print(df["content_age_days"].describe())

trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Content Types:
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64

Content Age:
count    30000.00000
mean       256.16780
std        132.70793
min         90.00000
25%        132.00000
50%        236.00000
75%        333.00000
max        564.00000
Name: content_age_days, dtype: float64


# Quick Look at the Data

The starter dataset contains **30,000 content pages**.

Some initial observations from the dataset are:

- **16,262 pages** are labeled with a **"down"** trend, making decline the largest trend category.
- The dataset is dominated by **keyword articles (27,207 pages)**, with smaller numbers of feedly articles (2,096) and comparison articles (697).
- The average content age is **256 days**, with all pages being at least **90 days old**, making the dataset suitable for analyzing content refresh opportunities.

These observations suggest that the dataset contains a large number of mature content pages with measurable performance trends, making it appropriate for investigating content refresh prioritization.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# Careful Words: What I Can and Can't Claim

This project is designed to support human decision-making rather than replace it. The ranked recommendations identify pages that may deserve earlier review based on observable search and engagement signals.

From this analysis, I cannot claim that refreshing a page will definitely improve its performance or that the model proves a cause-and-effect relationship. The results only show patterns observed in the starter dataset.

The findings are based on an anonymized starter dataset and should be validated on larger datasets before being used in real-world decision making. Human review is still required before taking any content action.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.